### PR TITLE
MYC-180

### PR DESCRIPTION
**Pull request by BtencateSphereon**
added ion-did-provider

### CODE REVIEW

File: packages/ion-did-provider/src/ion-did-resolver.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import * as ION from '@sphereon/ion-tools'
+
+export function IonDIDResolver() {
```
Comment (nklomp) [inline packages/ion-did-provider/src/ion-did-resolver.ts:3]: Hmzz, this should probably be more inline with the Resolver package from DIF, conforming to the Interface for resolution. Similar to what we have in the did-uni-client. Veramo also supports that interface for DID resolution

File: packages/ion-did-provider/__tests__/identity-provider.test.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+describe('dummy', () => {
+  const a = 100
+  it('should run a dummy test', () => {
```
Comment (nklomp) [inline packages/ion-did-provider/__tests__/identity-provider.test.ts:3]: Ideally we have 2 integration tests. For resolution and registration.

File: packages/ion-did-provider/src/ion-did-provider.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    { kms, options }: { kms?: string; options?: any },
+    context: IContext,
+  ): Promise<Omit<IIdentifier, 'provider'>> {
+    const key = await context.agent.keyManagerCreate({ kms: kms || this.defaultKms, type: 'Secp256k1' })
+    // convert to JWK (note, we should have a discussion about key dependencies here)
+    const publicJwk = secp256k1_keyUtils.publicKeyJwkFromPublicKeyHex(key.publicKeyHex)
+
+    const did = new ION.DID({
+      content: {
+        publicKeys: [
```
Comment (nklomp) [inline packages/ion-did-provider/src/ion-did-provider.ts:33]: Questions: Doesn;t it also need 2 other keys like for instance a recoveryKey?

**Review by nklomp — CHANGES_REQUESTED** [review]
Some remarks and a question

File: packages/ion-did-provider/src/ion-did-provider.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    const identifier: Omit<IIdentifier, 'provider'> = {
+      did: await did.getURI(),
+      controllerKeyId: key.kid,
+      keys: [key],
+      services: [],
+    }
+
+    // TODO: Long version DID is not conformant to "initial value" definition.
+    console.log(`Created (short version): ${ await did.getURI('short') }`)
```
[the remark this answers is not in the document]
Comment (BtencateSphereon) [reply packages/ion-did-provider/src/ion-did-provider.ts]: changed to debug

File: packages/ion-did-provider/src/ion-did-resolver.ts (hunk 1 above)
Comment (BtencateSphereon) [reply packages/ion-did-provider/src/ion-did-resolver.ts]: did some refactoring on the ion resolver

File: packages/ion-did-provider/src/ion-did-provider.ts (hunk 3 above)
Comment (BtencateSphereon) [reply packages/ion-did-provider/src/ion-did-provider.ts:33]: As discussed, created the following two tickets. Hopefully I wrote it down correctly

https://sphereon.atlassian.net/browse/WAL-204
https://sphereon.atlassian.net/browse/WAL-206

File: packages/ion-did-provider/src/ion-did-resolver.ts (hunk 1 above)
Comment (nklomp) [reply packages/ion-did-provider/src/ion-did-resolver.ts]: LGTM

**Review by nklomp — APPROVED** [review]
LGTM

**Pull request merged**